app/controllers/archives_controller.rb

```diff
@@ -12,6 +13,16 @@ def show
     render partial: 'archives/show', locals: { archive: archive }
   end
 
+  def batch
+    params[:archive_ids].each do |id|
+      if archive = Archive.find_by(id: id.first)
+        archive.transition_to :generating
+        CreateArchive.perform_async(archive.id)
```

Comment:
This is similar to the code in `update`, it should then be moved to a common method.